src/createRouter.ts

```diff
@@ -323,6 +375,118 @@ const createRouter: RouterFactory<Context> = compose.mixin(createEvented, {
 			}, cancel);
 		},
 
+		link(this: Router<Context>, route: Route<Context, Parameters>, params: LinkParams = {}): string {
+			const { history, routes: roots, currentSelection } = privateStateMap.get(this);
+
+			const hierarchy = [ route ];
+			for (let parent = route.parent; parent !== undefined; parent = parent.parent) {
+				hierarchy.unshift(parent);
+			}
+
+			if (!includes(roots, hierarchy[0])) {
+				throw new Error('Cannot generate link for route that is not in the hierarchy');
+			}
+
+			const { leadingSlash: addLeadingSlash } = hierarchy[0].path;
+			let addTrailingSlash = false;
+			const segments: string[] = [];
+			const searchParams = new UrlSearchParams();
+
+			type Zipped = {
+				currentPathValues?: string[];
+				currentSearchParams?: SearchParams;
+				path: DeconstructedPath;
+			}
+			hierarchy
+				.map<Zipped>((route, index) => {
```

Comment:
Is there a reason for the assertion here?  It seems logical that it would just be contextually inferred.